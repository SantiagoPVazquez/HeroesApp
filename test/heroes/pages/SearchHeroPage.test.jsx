import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter, useNavigate } from 'react-router-dom';
import { SearchHeroPage } from '../../../src/heroes';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}));

describe('Tests in <SearchHeroPage/>', () => { 

    beforeEach(() => jest.clearAllMocks() );
    
    test('should show correctly with default values', () => { 
        
        const {container} = render(
            <MemoryRouter>
                <SearchHeroPage/>
            </MemoryRouter>
        )
        
        expect(container).toMatchSnapshot();

     });

     test('should show hero was not found', () => { 

        render(
            <MemoryRouter initialEntries={['/search?q=wrongheroname']}>
                <SearchHeroPage/>
            </MemoryRouter>
        )

        expect(screen.getByText('was not found')).toBeTruthy();

      });

     test('should show Spider-man and input with queryString value', () => { 
        
        render(
            <MemoryRouter initialEntries={['/search?q=spider']}>
                <SearchHeroPage/>
            </MemoryRouter>
        )
        
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('spider');

        const img = screen.getByRole('img');
        expect(img.src).toContain('/assets/heroes/marvel-spider.jpg')

     });


     test('should call navigate to new page', () => {
        
        const inputValue = 'spider'
        
        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchHeroPage/>
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {name: 'searchText', value: inputValue}});

        const form = screen.getByLabelText('form')
        fireEvent.submit(form);
        
        expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
      });



 });