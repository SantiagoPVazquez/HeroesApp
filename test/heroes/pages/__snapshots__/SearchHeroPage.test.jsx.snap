// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <SearchHeroPage/> should show correctly with default values 1`] = `
<div>
  <h1>
    Search Hero
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Hero name"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeInUp"
      >
        Search a Hero
      </div>
    </div>
  </div>
</div>
`;
